src/App.tsx

```diff
@@ -1,26 +1,222 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodos,
+  deleteTodo,
+  getTodos,
+  patchTodos,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FormTodo } from './components/FormTodo';
+import { FooterTodos } from './components/FooterTodos';
+import { ErrorTodos } from './components/ErrorTodos';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterSelect, setFilterSelected] = useState<Filter>(Filter.All);
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+        throw new Error('Cant find todos');
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filterSelect === 'Active') {
+      return !todo.completed;
+    }
+
+    if (filterSelect === 'Completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  async function postTodos(title: string) {
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle.length === 0) {
+      setError('Title should not be empty');
+      inputRef.current?.focus();
+
+      return;
+    }
+
+    setIsDisabledInput(true);
+
+    const tempId = Date.now();
+
+    const tempTodo: Todo = {
+      id: tempId,
+      userId: 3177,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    setTodos(prev => [...prev, tempTodo]);
+
+    try {
+      const newTodo = await addTodos({
+        title: trimmedTitle,
+        completed: false,
+        userId: 3177,
+      });
+
+      setTodos(prev => prev.map(todo => (todo.id === tempId ? newTodo : todo)));
+      setSearchTerm('');
+    } catch {
+      setError('Unable to add a todo');
+      setTodos(prev => prev.filter(todo => todo.id !== tempId));
+      throw new Error('Cant create new todos');
+    } finally {
+      setIsDisabledInput(false);
+
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 0);
+    }
+  }
+
+  function removeTodos(todoId: number) {
```

Comment:
It is important to follow the same way of creating functions, fix it everywhere
```suggestion
  const removeTodos = (todoId: number) => {
```